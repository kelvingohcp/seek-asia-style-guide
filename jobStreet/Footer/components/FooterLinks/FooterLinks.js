import React from 'react';
import PropTypes from 'prop-types';
import _get from 'lodash/get';
import styles from './FooterLinks.less';

const ListLinks = ({ title, url }) => {
  if (title && url) {
    return (
      <li className={styles.item}>
        <a className={styles.link} href={url}>
          {title}
        </a>
      </li>
    );
  }
  return null;
};

const FooterLinks = ({ footerMessages }) => {
  return (
    <ul className={styles.list}>
      <ListLinks {..._get(footerMessages, 'link')} />
      <ListLinks {..._get(footerMessages, 'siteMap')} />
      <ListLinks {..._get(footerMessages, 'about')} />
      <ListLinks {..._get(footerMessages, 'profile')} />
      <ListLinks {..._get(footerMessages, 'terms')} />
      <ListLinks {..._get(footerMessages, 'privacy')} />
      <ListLinks {..._get(footerMessages, 'safe')} />
      <ListLinks {..._get(footerMessages, 'help')} />
      <ListLinks {..._get(footerMessages, 'feedback')} />
    </ul>
  );
};

FooterLinks.propTypes = {
  footerMessages: PropTypes.object.isRequired
};

ListLinks.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default FooterLinks;
