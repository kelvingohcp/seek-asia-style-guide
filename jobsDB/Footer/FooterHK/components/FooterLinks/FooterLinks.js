import React from 'react';
import PropTypes from 'prop-types';

import styles from './FooterLinks.less';
import { Text } from 'seek-asia-style-guide/react';

const FooterLinks = ({ messages, cfsLink, wordpressLink }) => {
  return (
    <div className={styles.list}>
      <div className={styles.item}>
        {wordpressLink(
          <Text className={styles.copyright} whistling secondary semiStrong>
            {messages['FooterHK.privacyLinkText']}
          </Text>,
          messages['FooterHK.privacyLink']
        )}
      </div>
      <div className={styles.item}>
        {wordpressLink(
          <Text className={styles.copyright} whistling secondary semiStrong>
            {messages['FooterHK.termsLinkText']}
          </Text>,
          messages['FooterHK.termsLink']
        )}
      </div>
      <div className={styles.item}>
        {cfsLink(
          <Text className={styles.copyright} whistling secondary semiStrong>
            {messages['FooterHK.siteMapLinkText']}
          </Text>,
          messages['FooterHK.siteMapLink']
        )}
      </div>
    </div>
  );
};

FooterLinks.propTypes = {
  messages: PropTypes.object.isRequired,
  cfsLink: PropTypes.func.isRequired,
  wordpressLink: PropTypes.func.isRequired
};

export default FooterLinks;
