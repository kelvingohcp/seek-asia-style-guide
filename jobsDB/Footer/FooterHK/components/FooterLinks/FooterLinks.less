@import (reference) "~seek-asia-style-guide/theme";
@import (reference) "../../FooterHK.less";

@list-line-height: @grid-base * 4;
@item-horizontal-padding: @grid-base * 2;

.list {
  display: block;
  padding-top: @grid-base * 2;
  padding-left: @grid-base * 3;
  @media @desktop {
    display: none;
  }
}

.displayInDesktop {
  display: none;
  padding: 0;
  @media @desktop {
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    display: flex;
    line-height: @list-line-height;
  }
}

.item {
  padding: 0;
  @media @desktop {
    padding: 0 @item-horizontal-padding;
  }
  &:first-child {
    padding-left: 0;
  }
}
