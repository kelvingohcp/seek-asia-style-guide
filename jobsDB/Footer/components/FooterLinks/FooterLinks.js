import React from 'react';
import PropTypes from 'prop-types';

import styles from './FooterLinks.less';

const FooterLinks = ({ messages }) => {
  return (
    <ul className={styles.list}>
      <li className={styles.itemMobile}>
        <a className={styles.link} href={messages['footer.legalLink']} rel="nofollow">
          {messages['footer.legalLinkText']}
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.privacyLink']} rel="nofollow">
          {messages['footer.privacyLinkText']}
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.termsLink']} rel="nofollow">
          {messages['footer.termsLinkText']}
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.siteMapLink']}>
          {messages['footer.siteMapLinkText']}
        </a>
      </li>
    </ul>
  );
};

FooterLinks.propTypes = {
  messages: PropTypes.object.isRequired
};

export default FooterLinks;
