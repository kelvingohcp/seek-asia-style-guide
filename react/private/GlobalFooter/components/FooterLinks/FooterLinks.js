import React from 'react';
import PropTypes from 'prop-types';

import classnames from 'classnames';
import styles from './FooterLinks.less';
import { Text } from 'seek-asia-style-guide/react';
import _get from 'lodash/get';

const FooterLinks = ({ footerMessages, linkRenderer, displayInDesktop }) => {
  const LinkItem = ({ title, url }) => {
    return (
      <div className={styles.item}>
        {
          linkRenderer({
            children: <Text whispering secondary semiStrong>
              {title}
            </Text>,
            href: url
          })
        }
      </div>
    );
  };

  LinkItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  };

  return (
    <div className={classnames(styles.list, displayInDesktop && styles.displayInDesktop)}>
      <LinkItem {..._get(footerMessages, 'privacy')} />
      <LinkItem {..._get(footerMessages, 'terms')} />
      <LinkItem {..._get(footerMessages, 'siteMap')} />
    </div>
  );
};

FooterLinks.propTypes = {
  footerMessages: PropTypes.object.isRequired,
  linkRenderer: PropTypes.func.isRequired,
  displayInDesktop: PropTypes.bool
};

export default FooterLinks;
