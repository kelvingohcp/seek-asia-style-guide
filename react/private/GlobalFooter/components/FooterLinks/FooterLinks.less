@import (reference) "~seek-asia-style-guide/theme";
@import (reference) "../../GlobalFooter.less";

@list-line-height: @grid-base * 4;
@item-horizontal-padding: @grid-base * 2;

.item {
  padding: 0;
}

.footerMeta {
  padding-right: @list-line-height;
  list-style-type: none;
  padding-top: @grid-base * 4;
  @media @aboveMobile {
    padding-top: 0;
    display: flex;
    align-items: center;
  }
}

.footerMetaItem {
  align-items: center;
  display: flex;
  margin-bottom: @grid-base * 2;
  @media @aboveMobile {
    margin-bottom: initial;
    margin-right: @grid-base * 3;
  }
}
