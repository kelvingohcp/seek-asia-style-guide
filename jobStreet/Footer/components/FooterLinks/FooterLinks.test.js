import React from 'react';
import { shallow } from 'enzyme';
import FooterLinks from './FooterLinks';

const baseUrl = 'https://jobstreet.com.my';
const baseUrlWithLanguage = `${baseUrl}/en`;

const footerMessages = {
  copyright: 'Copyright Â© {year} JobStreet.com',
  link: {
    title: 'JobStreet International',
    url: 'https://www.jobstreet.com'
  },
  siteMap: {
    title: 'Site Map',
    url: `${baseUrl}/sitemap.htm`
  },
  about: {
    title: 'About Us',
    url: `${baseUrlWithLanguage}/about-us/`
  },
  profile: {
    title: 'Work With Us',
    url: `${baseUrlWithLanguage}/career/JobStreet.htm`
  },
  terms: {
    title: 'Term of Use',
    url: `${baseUrlWithLanguage}/about-us/terms-of-use/`
  },
  privacy: {
    title: 'Privacy Policy',
    url: `${baseUrlWithLanguage}/about-us/privacy-policy/`
  },
  safe: {
    title: 'Safe Job Search Guide',
    url: `${baseUrlWithLanguage}/about-us/safe-job-search-guide/`
  },
  help: {
    title: 'Help',
    url: `${baseUrlWithLanguage}/user/`
  },
  feedback: {
    title: 'Send Feedback',
    url: 'https://myjobstreet.jobstreet.co.my/home/feedback.php?site=id&sub=feedback'
  }
};

describe('FooterLinks component', () => {
  it('should render supplied messages', () => {
    const wrapper = shallow(<FooterLinks footerMessages={footerMessages} />);
    expect(wrapper).toMatchSnapshot();
  });
});
