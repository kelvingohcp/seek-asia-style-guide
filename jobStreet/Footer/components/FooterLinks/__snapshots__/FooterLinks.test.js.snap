// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLinks component should render supplied messages 1`] = `
<ul
  className="list"
>
  <ListLinks
    title="JobStreet International"
    url="https://www.jobstreet.com"
  />
  <ListLinks
    title="Site Map"
    url="https://jobstreet.com.my/sitemap.htm"
  />
  <ListLinks
    title="About Us"
    url="https://jobstreet.com.my/en/about-us/"
  />
  <ListLinks
    title="Work With Us"
    url="https://jobstreet.com.my/en/career/JobStreet.htm"
  />
  <ListLinks
    title="Term of Use"
    url="https://jobstreet.com.my/en/about-us/terms-of-use/"
  />
  <ListLinks
    title="Privacy Policy"
    url="https://jobstreet.com.my/en/about-us/privacy-policy/"
  />
  <ListLinks
    title="Safe Job Search Guide"
    url="https://jobstreet.com.my/en/about-us/safe-job-search-guide/"
  />
  <ListLinks
    title="Help"
    url="https://jobstreet.com.my/en/user/"
  />
  <ListLinks
    title="Send Feedback"
    url="https://myjobstreet.jobstreet.co.my/home/feedback.php?site=id&sub=feedback"
  />
</ul>
`;
